src/webui/api/apicontroller.cpp

```diff
@@ -42,9 +42,9 @@ APIController::APIController(IApplication *app, QObject *parent)
 {
 }
 
-QVariant APIController::run(const QString &action, const StringMap &params, const DataMap &data)
+APIResult APIController::run(const QString &action, const StringMap &params, const DataMap &data)
 {
-    m_result.clear(); // clear result
+    m_result = {}; // clear result
```

Comment:
To make it equivalent, it should call `clear()` on each field otherwise it will allocate different memory region for each function call. `APIResult` perhaps can have a `clear()` method.

Reply:
> To make it equivalent, it should call `clear()` on each field otherwise it will allocate different memory region for each function call. `APIResult` perhaps can have a `clear()` method.

It makes sense. Done.